### PR TITLE
Task 2 get all topics

### PR DESCRIPTION
**Pull request by stevenwu99**
Task 2 get all topics
--Create app.js
--Create app.test.js
--Create app.controller.js
--Create app.model.js
--Create errors.js (error handler file)
--Using the test data for app.test.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+ //Task 2 CORE: GET /api/topics
+ describe ('GET /api/topics', () => {
+    test ('200:responses with all topics ', () => { 
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({body}) => {
+            const {topics} = body;  
+            expect(topics).toHaveLength(3);
```
Comment (johnhutcheon) [inline __tests__/app.test.js:25]: Nice work checking the length before a forEach 👍

File: db/error.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+exports.handleCustomErrors = (err,req,res,next) => {
+    if(err.msg) {
+        res.status(err.status).send({msg:err.msg});
+    } else next (err);
+};
+
+exports.handleServerErrors = (err,req,res,next) => {
+   res.status(500).send({msg:"Server broken..."});
+};
```
Comment (johnhutcheon) [inline db/error.js:3]: It's fine to have a 500 error handler at this stage, but do we need the other 2 error handlers? We should only be adding error handling when the tests you have written requires it. Please remove anything that's unnecessary. 👍

File: db/model/app.model.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+//Task 2 Get All topics
+exports.selectAllTopics = () => {
+    const selectSQLStr = "SELECT * FROM topics;";
+    return db
+    .query(selectSQLStr)
+    .then(({rows}) => {
+        if (rows.length === 0) {
+            return Promise.reject({status:404,msg:"Not found"})
+         };
```
Comment (johnhutcheon) [inline db/model/app.model.js]: As we've included this code to catch 404 errors, could we add a test for this?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Steven, nice work so far! Could you just see my comments please and remove any unnecessary tests? Always ensure the error handling you've implemented is required for the tests you've written. 👍 Please resubmit your Pull Request when you've amended this.

File: db/model/app.model.js (hunk 3 above)
Comment (stevenwu99) [reply db/model/app.model.js]: I think we don't need to include the 404 checks(not data found) for task 2. when I include 404 test , status 200 should return  error

File: db/error.js (hunk 2 above)
Comment (stevenwu99) [reply db/error.js:3]: I have removed  psql and custom error handler and keep server error handler.

**Review by Hard-Luck — APPROVED** [review]
This ticket it good to merge

**Pull request merged**